xclim/sdba/_adjustment.py

```diff
@@ -126,32 +141,315 @@ def eqm_train(
+
+    # npdf training core
+    af_q_l = []
+    escores_l = []
+
+    # loop over time blocks
+    for ib in range(gw_idxs[gr_dim].size):
+        # indices in a given time block
+        indices = gw_idxs[{gr_dim: ib}].astype(int).values
+        ind = indices[indices >= 0]
+
+        # npdft training : multiple rotations on standardized datasets
+        # keep track of adjustment factors in each rotation for later use
+        af_q, escores = xr.apply_ufunc(
+            _npdft_train,
+            ref[{"time": ind}],
+            hist[{"time": ind}],
+            rot_matrices,
+            quantiles,
+            input_core_dims=[
+                [pts_dims[0], "time"],
+                [pts_dims[0], "time"],
+                ["iterations", pts_dims[1], pts_dims[0]],
+                ["quantiles"],
+            ],
+            output_core_dims=[
+                ["iterations", pts_dims[1], "quantiles"],
+                ["iterations"],
+            ],
+            dask="parallelized",
+            output_dtypes=[hist.dtype, hist.dtype],
+            kwargs={
+                "method": interp,
+                "extrap": extrapolation,
+                "n_escore": n_escore,
+                "standardize": True,
+            },
+            vectorize=True,
+        )
+        af_q_l.append(af_q.expand_dims({gr_dim: [ib]}))
+        escores_l.append(escores.expand_dims({gr_dim: [ib]}))
+    af_q = xr.concat(af_q_l, dim=gr_dim)
+    escores = xr.concat(escores_l, dim=gr_dim)
+    out = xr.Dataset(dict(af_q=af_q, escores=escores)).assign_coords(
+        {"quantiles": quantiles, gr_dim: gw_idxs[gr_dim].values}
+    )
+    return out
+
+
+def _npdft_adjust(sim, af_q, rots, quantiles, method, extrap):
+    """Npdf transform adjusting.
+
+    Adjusting factors `af_q` obtained in the training step are applied on the simulated data `sim` at each iterated
+    rotation
+    """
+    # add dummy dim  if period_dim absent to uniformize the function below
+    # This could be done at higher level, not sure where is best
+    if dummy_dim_added := (len(sim.shape) == 2):
+        sim = sim[:, np.newaxis, :]
+
+    # adjust npdft
+    for ii in range(len(rots)):
+        rot = rots[0] if ii == 0 else rots[ii] @ rots[ii - 1].T
+        sim = np.einsum("ij,j...->i...", rot, sim)
+        # loop over variables
+        for iv in range(sim.shape[0]):
+            af = u._interp_on_quantiles_1D_multi(
+                u._rank_bn(sim[iv], axis=-1),
+                quantiles,
+                af_q[ii, iv],
+                method=method,
+                extrap=extrap,
+            )
+            sim[iv] = u.apply_correction(sim[iv], af, "+")
+
+    rot = rots[-1].T
+    sim = np.einsum("ij,j...->i...", rot, sim)
+    if dummy_dim_added:
+        sim = sim[:, 0, :]
+
+    return sim
+
+
+def mbcn_adjust(
+    ref,
+    hist,
+    sim,
+    ds,
+    pts_dims,
+    interp,
+    extrapolation,
+    base,
+    base_kws_vars,
+    adj_kws,
+    period_dim,
```

Comment:
Same comment about the annotations. The docstring also needs some numpydoc-ification I think. Section headers are missing.